objectRecognitionSource/src/test/java/io/vantiq/extsrc/objectRecognition/neuralNet/TestYoloProcessor.java

```diff
@@ -1217,7 +1222,432 @@ public void testQuery() throws ImageProcessingException, InterruptedException {
+        config.put("preCrop", preCrop);
+        checkInvalidPreCropping(config);
+        
+        // The coordinates are negative, should not do any resizing
+        preCrop.put("x", -1);
+        preCrop.put("y", -1);
+        preCrop.put("w", CROPPED_WIDTH);
+        preCrop.put("h", CROPPED_HEIGHT);
+        config.put("preCrop", preCrop); 
+        checkInvalidPreCropping(config);
+        
+        // Two values are not integers, should not do any resizing
+        preCrop.put("x", TOP_LEFT_X_COORDINATE);
+        preCrop.put("y", TOP_LEFT_Y_COORDINATE);
+        preCrop.put("w", 0.5);
+        preCrop.put("h", 0.5);
+        config.put("preCrop", preCrop);
+        checkInvalidPreCropping(config);
+        
+        // A value is not a number, should not do any resizing
+        preCrop.put("x", TOP_LEFT_X_COORDINATE);
+        preCrop.put("y", TOP_LEFT_Y_COORDINATE);
+        preCrop.put("w", "jibberish");
+        preCrop.put("h", CROPPED_HEIGHT);
+        config.put("preCrop", preCrop);
+        checkInvalidPreCropping(config);
+        
+        // The preCrop is not a map which is not allowed, should not do any resizing
+        config.put("preCrop", "jibberish");
+        checkInvalidPreCropping(config);
+    }
+    
+    // Helper method used to check invalid pre cropping configurations
+    public void checkInvalidPreCropping(Map config) throws ImageProcessingException, IOException {
+        YoloProcessor ypImageSaver = new YoloProcessor();
+        try {
+            ypImageSaver.setupImageProcessing(config, SOURCE_NAME, MODEL_DIRECTORY, testAuthToken, testVantiqServer);
+        } catch (Exception e) {
+            fail("Could not setup the YoloProcessor");
+        }
+        File d = new File(OUTPUT_DIR);
+        try {
+            // Ensure no results from previous tests
+            if (d.exists()) {
+                deleteDirectory(OUTPUT_DIR);
+            }
+
+            NeuralNetResults results = ypImageSaver.processImage(getTestImage());
+            assert results != null;
+            assert results.getResults() != null;
+
+            // Should saved image with timestamp
+            assert d.exists();
+            assert d.isDirectory();
+            assert d.listFiles().length == 1;
+            assert d.listFiles()[0].getName().matches(timestampPattern);
+            
+            File resizedImageFile = new File(OUTPUT_DIR + "/" + d.listFiles()[0].getName());
+            BufferedImage resizedImage = ImageIO.read(resizedImageFile);
+            
+            // Original dimensions
+            assert resizedImage.getWidth() == TEST_IMAGE_WIDTH;
+            assert resizedImage.getHeight() == TEST_IMAGE_HEIGHT;
+
+        } finally {
+            if (d.exists()) {
+                deleteDirectory(OUTPUT_DIR);
+            }
+
+            ypImageSaver.close();
+        }
+    }
+    
+    @Test
+    public void testPreCroppingLocal() throws ImageProcessingException, IOException {
```

Comment:
Seems like we're testing the "crop" part pretty thoroughly, but not the "pre" part.

It would be good to ensure that the image is actually cropped before analysis.  May have to expose the TF results to the test methods so that you can verify that either it doesn't find anything (since everything interesting was cropped out) or that what it does find is within the boundaries of the cropped image.

If the `crop()` method is exposed to the test, you may be able to reach in and find the last image dimensions or something?

Given the structure, not completely sure how to do this, but seems like it'd be good to verify that the analysis is running on something cropped -- not just that the parameter validation is OK and that the resulting image matches those...